### PR TITLE
don't parse image, for invalid server responses.

### PR DESCRIPTION
**Pull request by palamccc**

### CODE REVIEW

**Comment by palamccc** [comment]
I signed it!

**Comment by willnorris** [comment]
So I'm curious what the motivation for this is.  Were you running into specific trouble?

It's entirely possible that the remote server returns an image with a non-200 response.  In which case, transforming the image as requested still sort of makes sense.  If the response isn't an image, then it obviously won't be transformed anyway.

File: imageproxy.go [hunk 1]
```diff
@@ -283,11 +283,16 @@ func (t *TransformingTransport) RoundTrip(req *http.Request) (*http.Response, er
 		return nil, err
 	}
 
-	opt := ParseOptions(req.URL.Fragment)
-
-	img, err := Transform(b, opt)
-	if err != nil {
-		glog.Errorf("error transforming image: %v", err)
+	var img []byte;
+	if resp.StatusCode == 200 {
```
Comment (kazhuravlev) [inline imageproxy.go]:
```
if resp.StatusCode == http.StatusOK {
...
}
```

**Comment by willnorris** [comment]
Closing this PR, since there was never any response to why this is actually needed.  (Please feel free to continue the discussion if this is still needed)

@huguesalary's https://github.com/betabrand/imageproxy/commit/34252f774807deab587908dec5813fa236fb3a2f seemed to be related to this (though I'm not 100% sure how).  That has now been merged as 3444fd9cb479db44758600de90f0f60dd1e6970e, so maybe that fixes the root issue anyway?

**Pull request closed without merging**